drivers/mbox/mbox_nxp_imx_mu.c

```diff
@@ -61,7 +61,13 @@ static int nxp_imx_mu_send(const struct device *dev, uint32_t channel, const str
 
 	/* Signalling mode. */
 	if (msg == NULL) {
-		return MU_TriggerInterrupts(cfg->base, g_gen_int_trig_mask[channel]);
+		int ret = MU_TriggerInterrupts(cfg->base, g_gen_int_trig_mask[channel]);
+
+		if (ret != 0) {
+			/* interrupt already pending, cannot trigger again */
+			return -EAGAIN;
+		}
+		return ret;
```

Comment:
nit, could be simplified to:
```suggestion
		if(MU_TriggerInterrupts(cfg->base, g_gen_int_trig_mask[channel]) != 0) {
			/* interrupt already pending, cannot trigger again */
			return -EAGAIN;
		}
		return 0;
```
or even
```suggestion
                return (MU_TriggerInterrupts(cfg->base, g_gen_int_trig_mask[channel]) != 0) ? -EAGAIN : 0;
```
(although the latter, while a one-liner, might be harder to look at, so maybe the first is better)

Reply:
Updated with the first suggestion.